### PR TITLE
Calculate HMAC of token parameters before using them for calculation

### PR DESCRIPTION
**Pull request by inorick**

### CODE REVIEW

File: northstar/src/runtime/token.rs [hunk 1]
```diff
@@ -96,17 +96,19 @@ fn now() -> time::Duration {
...
+    hmac.update(user);
+    let user = hmac.finalize_reset();
+    hmac.update(target);
+    let target = hmac.finalize_reset();
+    hmac.update(shared);
+    let shared = hmac.finalize_reset();
+    hmac.update(&time.as_millis().to_be_bytes());
+    hmac.update(&user.into_bytes());
+    hmac.update(&target.into_bytes());
+    hmac.update(&shared.into_bytes());
```
Comment (flxo) [inline northstar/src/runtime/token.rs:110]: nit: The `into_bytes` call should not be needed.

**Review by flxo — APPROVED** [review]

File: northstar/src/runtime/token.rs (hunk 1 above)
Comment (inorick) [reply northstar/src/runtime/token.rs:110]: I thought so too but omitting it gives
```
hmac.update(&user);
            ^^^^^ expected slice `[u8]`, found struct `sha2::digest::CtOutput
```

Comment (flxo) [reply northstar/src/runtime/token.rs:110]: `user.as_ref()`?

Comment (inorick) [reply northstar/src/runtime/token.rs:110]: Does not exists. The API of CtOutput is very restricted and does not offer much more than `new` and `into_bytes`:
https://docs.rs/digest/latest/x86_64-apple-darwin/digest/struct.CtOutput.html

The HMAC docs state that `To get underlying array use into_bytes ...`
https://docs.rs/hmac/0.12.1/hmac/

Comment (flxo) [reply northstar/src/runtime/token.rs:110]: Ah ok. Sorry I didn't realise that `user` and `target` is a `CtOutput` not the arg type `AsRef<[u8]>`.

**Pull request merged**